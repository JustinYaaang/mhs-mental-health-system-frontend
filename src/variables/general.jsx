// ##############################
// // // Tasks for TasksCard - see Dashboard view
// #############################

var bugs = [
  'Sign contract for "What are conference organizers afraid of?"',
  'Lines From Great Russian Literature? Or E-mails From My Boss?',
  'Flooded: One year later, assessing what was lost and what was found when a ravaging rain swept through metro Detroit',
  'Create 4 Invisible User Experiences you Never Knew About'
]
var website = [
  'Flooded: One year later, assessing what was lost and what was found when a ravaging rain swept through metro Detroit',
  'Sign contract for "What are conference organizers afraid of?"'
]
var server = [
  'Lines From Great Russian Literature? Or E-mails From My Boss?',
  'Flooded: One year later, assessing what was lost and what was found when a ravaging rain swept through metro Detroit',
  'Sign contract for "What are conference organizers afraid of?"'
]

var baseUrl = 'http://localhost:3000/api/v1/'//'http://178.128.34.125/api/v1/'//'http://127.0.0.1:3000/api/v1/'// 'http://mhsbackend.azurewebsites.net/api/v1/'
var fetchQuestionnairesUrl = 'questionnaire_sJS'
var patientanswersUrl = 'patientanswers'
var backendUrl = 'http://178.128.34.125/api/v1/'
var createUserAnswers = 'useranswers'
var authenticationUrl = 'patients/authenticate'
var questionnaireWithoutToken = 'questionnaires'
var Organizations = 'Organisations'

module.exports = {
  // these 3 are used to create the tasks lists in TasksCard - Dashboard view
  bugs,
  website,
  server,
  baseUrl,
  backendUrl,
  fetchQuestionnairesUrl,
  patientanswersUrl,
  createUserAnswers,
  authenticationUrl,
  questionnaireWithoutToken,
  Organizations
}
