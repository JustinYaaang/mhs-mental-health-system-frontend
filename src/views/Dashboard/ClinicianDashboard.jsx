import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react plugin for creating charts
import ChartistGraph from "react-chartist";
// @material-ui/core
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import DateRange from "@material-ui/icons/DateRange";
import LocalOffer from "@material-ui/icons/LocalOffer";
import Update from "@material-ui/icons/Update";
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import AccessTime from "@material-ui/icons/AccessTime";
import Accessibility from "@material-ui/icons/Accessibility";
import Grade from "@material-ui/icons/Grade";
import Code from "@material-ui/icons/Code";
import All from "@material-ui/icons/AllInboxOutlined";
// core components
import GridItem from "components/Grid/GridItem.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import TaskView from "components/Tasks/TaskView.jsx";
import CustomTabs from "components/CustomTabs/CustomTabs.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardIcon from "components/Card/CardIcon.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardFooter from "components/Card/CardFooter.jsx";
import swal from 'sweetalert2';
import InformationCard from 'components/DashboardComponent/InformationCard.jsx';
import LineGraph from 'components/DashboardComponent/LineGraph.jsx';

import dashboardStyle from "assets/jss/material-dashboard-react/views/dashboardStyle.jsx";
import { fetchQuestionnaires, deleteQuestionnaire, fetchWeeklyResult,fetchUserAnswers } from "../../services/BackendService";

class Dashboard extends React.Component {
  state = {
    value: 0,
    idPublishedList: [],
    questionnairePublishedList: [],
    idDraftList: [],
    questionnaireDraftList: [],
    totalQuestionnaire: 0,
    totalPending:0,
    totalClose:0,
    dailySubmission: {
      labels:[],
      series:[[]]
    },
    seriesMax: 0,
  };
  handleChange = (event, value) => {
    this.setState({ value: value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleViewQuestionnaireClick = (index, status) => {
  
    if(status === 'DRAFT'){
      const questionnaireId = this.state.idDraftList[index];
      this.props.history.push(this.props.history.location.pathname + "/questionnaire/" + questionnaireId)

    }
    else if(status === 'PUBLISHED'){
      const questionnaireId = this.state.idPublishedList[index];
      this.props.history.push(this.props.history.location.pathname + "/questionnaire/" + questionnaireId)

    }
  };

  timeTrans(date){
    date = new Date(date);
    var Y = date.getFullYear() + '-';
    var M = (date.getMonth()+1 < 10 ? '0'+(date.getMonth()+1) : date.getMonth()+1) + '-';
    var D = (date.getDate() < 10 ? '0' + (date.getDate()) : date.getDate()) + ' ';
    return Y+M+D;
  };

  componentWillMount() {

    var todayTime = new Date( Date.parse( new Date()));
    var todayDate = this.timeTrans(todayTime);
    var lastTime = new Date( Date.parse( new Date())-(7*86400000));
    var lastDate = this.timeTrans(lastTime);
    var period = {
      "todayDate": todayDate,
      "lastDate": lastDate
    }

    fetchWeeklyResult(todayDate,lastDate).then(
      response => {
        var dailysubmit = {
          labels: [],
          series: [[]]
        }

        var dataTransfer = new Map([[0, 'S'], [1, 'M'], [2,'T'],[3, 'W'], [4, 'T'], [5, 'F'], [6,'S']]);
        var myday=todayTime.getDay();
        
        for(var i =6; i>= 0; i--){
          if(myday == -1){
            myday = 6;
          }
          if(response[myday] === undefined){
            dailysubmit.labels[i] = dataTransfer.get(myday);
            dailysubmit.series[0][i] = 0;
          }
          else{
            dailysubmit.labels[i] = dataTransfer.get(myday);
            dailysubmit.series[0][i] = response[myday].count;
          }   
          myday--;
      }
        console.log(dailysubmit.series[0]);
        console.log(Math.max(...dailysubmit.series[0]));
        var seriesMax = 1.10 * Math.max(...dailysubmit.series[0]);
        this.setState({'dailySubmission':dailysubmit, 'seriesMax': seriesMax})
      },
    );
    
    fetchQuestionnaires().then( //!!! AWAIT HERE
       response => {
        this.setState({'totalQuestionnaire': response.idPublishedList.length + response.questionnaireDraftList.length,
        'idDraftList': response.idDraftList, 'idPublishedList': response.idPublishedList, 
              'questionnaireDraftList': response.questionnaireDraftList, 'questionnairePublishedList': response.questionnairePublishedList});
      }
    );

    fetchUserAnswers()
      .then(response => {
        var pending = 0;
        var close = 0;
        for(var i = 0; i <  response.length; i++){
            if(response[i].status == 'PENDING'){
                pending++;
            }
            else{
                close++;
            }
        }
       
        this.setState({ totalPending:pending,totalClose:close })
      })
      .catch(error => {
        console.error(error)
      })
  }

  render() {
    var Chartist = require("chartist");
    const { classes } = this.props;

    const dashboardData = {
      dailySalesChart: {
        options: {
          lineSmooth: Chartist.Interpolation.cardinal({
            tension: 0
          }),
          low: 0,
          high: this.state.seriesMax, // creative tim: we recommend you to set the high sa the biggest value + something for a better look
          chartPadding: {
            top: 0,
            right: 0,
            bottom: 0,
            left: 0
          }
        }
      },

      unanswered: 8,
      percentage: 50,
    }

    return (
      <div>
        <GridContainer>
          <InformationCard 
          color={"info"} title={"Total Questionnaires"} value={this.state.totalQuestionnaire}
          daterange={"Updated today"} classes={classes}
          />
          
          <InformationCard 
          color={"danger"} title={"Total Pending Cases"} value={this.state.totalPending}
          daterange={"Just updated"} classes={classes}
          />
          <InformationCard 
          color={"success"} title={"Total Close Cases"} value={this.state.totalClose}
          daterange={"Updated today"} classes={classes}
          />

        </GridContainer>
  
        <GridContainer>
          <LineGraph
          color={"success"} dailySubmission={this.state.dailySubmission} type={"Line"}
          dashboardData={dashboardData}
          classes={classes}
          />
       

          <GridItem xs={12} sm={12} md={8}>
            <CustomTabs
              title="Questionnaires :"
              headerColor="info"
              onCreateNewClicked={() => this.handleCreateNewQuestionnaireClicked()}
              tabs={[
                {
                  tabName: "PUBLISHED",
                  tabIcon: Grade,
                  tabContent: (
                    <TaskView
                      tableHeaderColor="info"
                      tableHead={["Name", "Description", "Status", "Modify"]}
                      checkedIndexes={[]}
                      tasks={this.state.questionnairePublishedList}
                      onViewClicked={(index) => this.handleViewQuestionnaireClick(index, 'PUBLISHED')}
                    />
                  )
                },
                { 
                  tabName: "DRAFT",
                  tabIcon: Code,
                  tabContent: (
                    <TaskView
                      tableHeaderColor="primary"
                      tableHead={["Name", "Description", "Status", "Modify"]}
                      checkedIndexes={[]}
                      tasks={this.state.questionnaireDraftList}
                      onViewClicked={(index) => this.handleViewQuestionnaireClick(index, 'DRAFT')}
                    />
                  )
                }, 
              ]}
            />
          </GridItem>
        </GridContainer>
      </div>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(dashboardStyle)(Dashboard);
