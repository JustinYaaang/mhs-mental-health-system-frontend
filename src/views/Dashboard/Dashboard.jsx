import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react plugin for creating charts
import ChartistGraph from "react-chartist";
// @material-ui/core
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import DateRange from "@material-ui/icons/DateRange";
import LocalOffer from "@material-ui/icons/LocalOffer";
import Update from "@material-ui/icons/Update";
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import AccessTime from "@material-ui/icons/AccessTime";
import Accessibility from "@material-ui/icons/Accessibility";
import Grade from "@material-ui/icons/Grade";
import Code from "@material-ui/icons/Code";
import All from "@material-ui/icons/AllInboxOutlined";
// core components
import GridItem from "components/Grid/GridItem.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import Tasks from "components/Tasks/Tasks.jsx";
import AdminTabs from "components/CustomTabs/AdminTabs.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardIcon from "components/Card/CardIcon.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardFooter from "components/Card/CardFooter.jsx";
import swal from 'sweetalert';
import InformationCard from 'components/DashboardComponent/InformationCard.jsx';
import LineGraph from 'components/DashboardComponent/LineGraph.jsx';

import dashboardStyle from "assets/jss/material-dashboard-react/views/dashboardStyle.jsx";
import { fetchQuestionnaires, deleteQuestionnaire, fetchWeeklyResult } from "../../services/BackendService";

function getRole(){
  console.log(sessionStorage.role);
  return sessionStorage.role;
}

class Dashboard extends React.Component {
  state = {
    value: 0,
    idPublishedList: [],
    questionnairePublishedList: [],
    idDraftList: [],
    questionnaireDraftList: [],
    totalQuestionnaire: 0,
    dailySubmission: {
      labels:[],
      series:[[]]
    },
    seriesMax: 0,
  };
  handleChange = (event, value) => {
    this.setState({ value: value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleEditQuestionnaireClick = (index, status) => {
    console.log(index);
    if(status === 'DRAFT'){
      const questionnaireId = this.state.idDraftList[index];
      document.location.href = "/questionnaire/" + questionnaireId;
    }
    else if(status === 'PUBLISHED'){
      const questionnaireId = this.state.idPublishedList[index];
      document.location.href = "/questionnaire/" + questionnaireId;
    }
  };

  handleDeleteQuestionnaireClick = (index, status) => {
    swal({
      title: "Are you sure?",
      text: "The questionnaire cannot recover!",
      icon: "warning",
      buttons: true,
      dangerMode: true,
    })
    .then((willDelete) => {
      if (willDelete) {
        swal("The questionnaire has been deleted!", {
          icon: "success",
        });

        if(status === 'DRAFT'){
          const questionnaireId = this.state.idDraftList[index];
          deleteQuestionnaire(questionnaireId).then(
            
            response => {
              const idListBuffer = this.state.idDraftList.slice();
              idListBuffer.splice(index, 1);
              const questionnaireListBuffer = this.state.questionnaireDraftList.slice();
              questionnaireListBuffer.splice(index, 1);
              this.setState({idDraftList: idListBuffer, questionnaireDraftList: questionnaireListBuffer });
            }
          );
        }
        else if(status === 'PUBLISHED'){
          const questionnaireId = this.state.idPublishedList[index];
          deleteQuestionnaire(questionnaireId).then(
            response => {
              const idListBuffer = this.state.idPublishedList.slice();
              idListBuffer.splice(index, 1);
              const questionnaireListBuffer = this.state.questionnairePublishedList.slice();
              questionnaireListBuffer.splice(index, 1);
              this.setState({idPublishedList: idListBuffer, questionnairePublishedList: questionnaireListBuffer });
            }
          );
        }
      } else {
        swal("The questionnaire is safe!");
      }
    });
  };

  handleCreateNewQuestionnaireClicked = () => {
    document.location.href = "/questionnaire/";
  };

  timeTrans(date){
    date = new Date(date);
    var Y = date.getFullYear() + '-';
    var M = (date.getMonth()+1 < 10 ? '0'+(date.getMonth()+1) : date.getMonth()+1) + '-';
    var D = (date.getDate() < 10 ? '0' + (date.getDate()) : date.getDate()) + ' ';
    return Y+M+D;
  };

  componentWillMount() {

    getRole();

    var todayTime = new Date( Date.parse( new Date()));
    var todayDate = this.timeTrans(todayTime);
    var lastTime = new Date( Date.parse( new Date())-(7*86400000));
    var lastDate = this.timeTrans(lastTime);
    var period = {
      "todayDate": todayDate,
      "lastDate": lastDate
    }

    fetchWeeklyResult(todayDate,lastDate).then(
      response => {
        var dailysubmit = {
          labels: [],
          series: [[]]
        }

        var dataTransfer = new Map([[0, 'S'], [1, 'M'], [2,'T'],[3, 'W'], [4, 'T'], [5, 'F'], [6,'S']]);
        var myday=todayTime.getDay();
        
        for(var i =6; i>= 0; i--){
          if(myday == -1){
            myday = 6;
          }
          if(response[myday] === undefined){
            dailysubmit.labels[i] = dataTransfer.get(myday);
            dailysubmit.series[0][i] = 0;
          }
          else{
            dailysubmit.labels[i] = dataTransfer.get(myday);
            dailysubmit.series[0][i] = response[myday].count;
          }   
          myday--;
      }
        console.log(dailysubmit.series[0]);
        console.log(Math.max(...dailysubmit.series[0]));
        var seriesMax = 1.10 * Math.max(...dailysubmit.series[0]);
        this.setState({'dailySubmission':dailysubmit, 'seriesMax': seriesMax})
      },
    );
    
     fetchQuestionnaires().then( //!!! AWAIT HERE
       response => {
        this.setState({'totalQuestionnaire': response.idPublishedList.length + response.questionnaireDraftList.length,
        'idDraftList': response.idDraftList, 'idPublishedList': response.idPublishedList, 
              'questionnaireDraftList': response.questionnaireDraftList, 'questionnairePublishedList': response.questionnairePublishedList});
      }
    );
  }


  

  render() {
    var Chartist = require("chartist");
    const { classes } = this.props;

    const dashboardData = {
      dailySalesChart: {
        options: {
          lineSmooth: Chartist.Interpolation.cardinal({
            tension: 0
          }),
          low: 0,
          high: this.state.seriesMax, // creative tim: we recommend you to set the high sa the biggest value + something for a better look
          chartPadding: {
            top: 0,
            right: 0,
            bottom: 0,
            left: 0
          }
        }
      },

      unanswered: 8,
      waiting_patients: 18,
      percentage: 50,
    }

    console.log(dashboardData)
    console.log(this.state.dailySubmission)
    return (
      <div>
        <GridContainer>
          <InformationCard 
          color={"info"} title={"Total Questionnaires"} value={this.state.totalQuestionnaire}
          daterange={"Updated today"} classes={classes}
          />
          <InformationCard 
          color={"danger"} title={"Pending Cases"} value={dashboardData.unanswered}
          daterange={"Updated just now"} classes={classes}
          />

            <InformationCard 
          color={"success"} title={"Total Cases"} value={dashboardData.waiting_patients}
          daterange={"Just updated"} classes={classes}
          />

        </GridContainer>
        <GridContainer>
          <LineGraph
          color={"success"} dailySubmission={this.state.dailySubmission} type={"Line"}
          dashboardData={dashboardData}
          classes={classes}
          
          />
          <GridItem xs={12} sm={12} md={8}>
            <AdminTabs
              title="Questionnaires :"
              headerColor="info"
              onCreateNewClicked={() => this.handleCreateNewQuestionnaireClicked()}
              tabs={[
                {
                  tabName: "PUBLISHED",
                  tabIcon: Grade,
                  tabContent: (
                    <Tasks
                      tableHeaderColor="info"
                      tableHead={["Name", "Description", "Status", "Modify"]}
                      checkedIndexes={[]}
                      tasks={this.state.questionnairePublishedList}
                      onEditClicked={(index) => this.handleEditQuestionnaireClick(index, 'PUBLISHED')}
                      onDeleteClicked={(index) => this.handleDeleteQuestionnaireClick(index, 'PUBLISHED')}
                    />
                  )
                },
                { 
                  tabName: "DRAFT",
                  tabIcon: Code,
                  tabContent: (
                    <Tasks
                      tableHeaderColor="primary"
                      tableHead={["Name", "Description", "Status", "Modify"]}
                      checkedIndexes={[]}
                      tasks={this.state.questionnaireDraftList}
                      onEditClicked={(index) => this.handleEditQuestionnaireClick(index, 'DRAFT')}
                      onDeleteClicked={(index) => this.handleDeleteQuestionnaireClick(index, 'DRAFT')}
                    />
                  )
                }, 
              ]}
            />
          </GridItem>

        </GridContainer>
      </div>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(dashboardStyle)(Dashboard);
