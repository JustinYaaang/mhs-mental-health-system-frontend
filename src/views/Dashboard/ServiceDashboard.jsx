import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react plugin for creating charts
import ChartistGraph from "react-chartist";
// @material-ui/core
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import DateRange from "@material-ui/icons/DateRange";
import LocalOffer from "@material-ui/icons/LocalOffer";
import Update from "@material-ui/icons/Update";
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import AccessTime from "@material-ui/icons/AccessTime";
import Accessibility from "@material-ui/icons/Accessibility";
import Grade from "@material-ui/icons/Grade";
import Code from "@material-ui/icons/Code";
import All from "@material-ui/icons/AllInboxOutlined";
// core components
import GridItem from "components/Grid/GridItem.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import Tasks from "components/Tasks/Tasks.jsx";
import TaskView from "components/Tasks/TaskView.jsx";
import CustomTabs from "components/CustomTabs/CustomTabs.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardIcon from "components/Card/CardIcon.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardFooter from "components/Card/CardFooter.jsx";
import InformationCard from 'components/DashboardComponent/InformationCard.jsx';
import LineGraph from 'components/DashboardComponent/LineGraph.jsx';

import dashboardStyle from "assets/jss/material-dashboard-react/views/dashboardStyle.jsx";
import {fetchUserAnswers, fetchWeeklyResult } from "../../services/BackendService";
import Questionnaire from "../Questionnaire/Questionnaire.jsx"
import LineChart from "../LineChart/LineChart"
class Dashboard extends React.Component {
  state = {
    value: 0,
    totalStep: 0,
    totalPending:0,
    totalClose:0,
  };
  handleChange = (event, value) => {
    this.setState({ value: value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleViewResultClick = (page) => {
    if(page == 'triagelist'){
        window.location.href = '/service/triagelist'
    }
    else{
        console.log("clickkkkkkk")
    }
  };

  componentWillMount() {
    fetchUserAnswers()
      .then(response => {
        var rowsPending = 0;
        var rowsResolve = 0;

        for (var i = 0; i < response.length; i++) {
            if(response[i].status == 'PENDING'){ //&& response[i].service ==serviceClinician ){
              rowsPending++;
            }
            else if (response[i].status == 'RESOLVED'){// && response[i].service ==serviceClinician ){
              rowsResolve++;
            }
          }
        this.setState({ 'totalPending': rowsPending,'totalClose': rowsResolve })
      })
      .catch(error => {
        console.error(error)
      })
  }

  render() {

    const { classes } = this.props;

    return (
      <div>
        <GridContainer>
          <InformationCard 
            color={"info"} title={"Outstanding Cases"} value={this.state.totalPending}
            daterange={"Updated today"} classes={classes} onViewClicked={() => this.handleViewResultClick('triagelist')}
          />
          
          <InformationCard 
            color={"danger"} title={"Number Triaged"} value={this.state.totalClose}
            daterange={"Just updated"} classes={classes} onViewClicked={() => this.handleViewResultClick('triagelist')}
          />

          <InformationCard 
            color={"success"} title={"More Information Required"} value={this.state.totalStep}
            daterange={"Updated today"} classes={classes} onViewClicked={() => this.handleViewResultClick('triagelist')}
          />

        </GridContainer>

        <GridContainer>
          <LineChart/>
          <Questionnaire question = {this.props.history} value = {8}/>
        </GridContainer>
   
      </div>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(dashboardStyle)(Dashboard);
