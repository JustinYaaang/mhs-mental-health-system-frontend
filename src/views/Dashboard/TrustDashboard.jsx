import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react plugin for creating charts
import ChartistGraph from "react-chartist";
// @material-ui/core
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import DateRange from "@material-ui/icons/DateRange";
import LocalOffer from "@material-ui/icons/LocalOffer";
import Update from "@material-ui/icons/Update";
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import AccessTime from "@material-ui/icons/AccessTime";
import Accessibility from "@material-ui/icons/Accessibility";
import Grade from "@material-ui/icons/Grade";
import Code from "@material-ui/icons/Code";
import All from "@material-ui/icons/AllInboxOutlined";
// core components
import GridItem from "components/Grid/GridItem.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import TaskView from "components/Tasks/TaskView.jsx";
import CustomTabs from "components/CustomTabs/CustomTabs.jsx";
import InformationCard from 'components/DashboardComponent/InformationCard.jsx';
import LineGraph from 'components/DashboardComponent/LineGraph.jsx';

import dashboardStyle from "assets/jss/material-dashboard-react/views/dashboardStyle.jsx";
import { fetchQuestionnaires, getOrganizations, fetchWeeklyResult } from "../../services/BackendService";
import Questionnaire from "../Questionnaire/Questionnaire.jsx"
import LineChart from "../LineChart/LineChart"

class Dashboard extends React.Component {
  state = {
    value: 0,
    totalPublishedQuestionnaire: 0,
    totalDraftQuestionnaire: 0,
    totalService: 0,
  };

  handleChange = (event, value) => {
    this.setState({ value: value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleViewResultClick = (page) => {
    if(page == 'questionnaire'){
        window.location.href = '/trust/questionnaire'
    }
    else if(page == 'service'){
        window.location.href = '/trust/service'
    }
  };

  componentWillMount() {

    fetchQuestionnaires().then( //!!! AWAIT HERE
      response => {
        this.setState({'totalPublishedQuestionnaire': response.idPublishedList.length,  'totalDraftQuestionnaire':response.questionnaireDraftList.length,
        });
      }
    );

    getOrganizations().then(response=>{
      this.setState({'totalService':response.length})
    })
  }

  render() {
    const { classes } = this.props;

    return (
      <div>
        <GridContainer>
          <InformationCard 
            color={"info"} title={"Total Published Questionnaire"} value={this.state.totalPublishedQuestionnaire}
            daterange={"Updated today"} classes={classes} onViewClicked={() => this.handleViewResultClick('questionnaire')}
          />
          <InformationCard 
            color={"success"} title={"Total Draft Questionnaire"} value={this.state.totalDraftQuestionnaire}
            daterange={"Updated just now"} classes={classes} onViewClicked={() => this.handleViewResultClick('questionnaire')}
          />

          <InformationCard 
            color={"primary"} title={"Total Services"} value={this.state.totalService}
            daterange={"Updated just now"} classes={classes} onViewClicked={() => this.handleViewResultClick('service')}
          />

        </GridContainer>
        <GridContainer>
          <LineChart/>
          <Questionnaire question = {this.props.history} value = {8}/>
        </GridContainer>
      </div>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(dashboardStyle)(Dashboard);
