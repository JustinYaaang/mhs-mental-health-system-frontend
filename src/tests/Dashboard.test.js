import React from 'react'
import { shallow, mount } from 'enzyme'
import Dashboard from '../views/Dashboard/Dashboard'
import Trust from '../views/Dashboard/TrustDashboard'
import Service from '../views/Dashboard/ServiceDashboard'
import Clinician from '../views/Dashboard/ClinicianDashboard'
import { baseUrl, backendURL, fetchQuestionnairesUrl, patientanswersUrl, authenticationUrl, questionnaireWithoutToken } from '../variables/general'
import TrustDashboard from 'views/Dashboard/TrustDashboard';
import ServiceDashboard from 'views/Dashboard/ServiceDashboard';
import ClinicianDashboard from 'views/Dashboard/ClinicianDashboard';

describe('Admin Dashboard', () => {
  const nock = require('nock')

  const scope = nock(baseUrl + fetchQuestionnairesUrl)
    .get('/repos/atom/atom/license')
    .reply(200, {
      'message': 'SurveyJS questionnaires retrieved successfully',
      'data': [
        {
          '_id': '5d1a1d16d910160030d04979',
          'title': 'Triage To Refer',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1b3143cd1699002fd7b6cd',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'DRAFT'
        },
        {
          '_id': '5d1b655ad01e83503e3a6e55',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1c95a173589a0030d798af',
          'title': 'Test',
          'description': 'Test questionnaire',
          'status': 'DRAFT'
        }
      ]
    })

  it('should render correctly', async () => {
    const component = mount(<Dashboard />)

    expect(component).toMatchSnapshot()
  })
})


describe('Trust Dashboard', () => {
  const nock = require('nock')

  const scope = nock(baseUrl + fetchQuestionnairesUrl)
    .get('/repos/atom/atom/license')
    .reply(200, {
      'message': 'SurveyJS questionnaires retrieved successfully',
      'data': [
        {
          '_id': '5d1a1d16d910160030d04979',
          'title': 'Triage To Refer',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1b3143cd1699002fd7b6cd',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'DRAFT'
        },
        {
          '_id': '5d1b655ad01e83503e3a6e55',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1c95a173589a0030d798af',
          'title': 'Test',
          'description': 'Test questionnaire',
          'status': 'DRAFT'
        }
      ]
    })

  it('should render correctly', async () => {
    const component = mount(<TrustDashboard />)

    expect(component).toMatchSnapshot()
  })
})


describe('Service Dashboard', () => {
  const nock = require('nock')

  const scope = nock(baseUrl + fetchQuestionnairesUrl)
    .get('/repos/atom/atom/license')
    .reply(200, {
      'message': 'SurveyJS questionnaires retrieved successfully',
      'data': [
        {
          '_id': '5d1a1d16d910160030d04979',
          'title': 'Triage To Refer',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1b3143cd1699002fd7b6cd',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'DRAFT'
        },
        {
          '_id': '5d1b655ad01e83503e3a6e55',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1c95a173589a0030d798af',
          'title': 'Test',
          'description': 'Test questionnaire',
          'status': 'DRAFT'
        }
      ]
    })

  it('should render correctly', async () => {
    const component = mount(<ServiceDashboard />)

    expect(component).toMatchSnapshot()
  })
})



describe('Clinician Dashboard', () => {
  const nock = require('nock')

  nock(baseUrl + fetchQuestionnairesUrl)
    .get('/repos/atom/atom/license')
    .reply(200, {
      'message': 'SurveyJS questionnaires retrieved successfully',
      'data': [
        {
          '_id': '5d1a1d16d910160030d04979',
          'title': 'Triage To Refer',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1b3143cd1699002fd7b6cd',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'DRAFT'
        },
        {
          '_id': '5d1b655ad01e83503e3a6e55',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1c95a173589a0030d798af',
          'title': 'Test',
          'description': 'Test questionnaire',
          'status': 'DRAFT'
        }
      ]
    })

  nock(baseUrl + patientanswersUrl)
    .get('/repos/atom/atom/license')
    .reply(200, {
      'message': 'SurveyJS questionnaires retrieved successfully',
      'data': [
        {
          '_id': '5d1a1d16d910160030d04979',
          'title': 'Triage To Refer',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1b3143cd1699002fd7b6cd',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'DRAFT'
        },
        {
          '_id': '5d1b655ad01e83503e3a6e55',
          'title': 'Triage To Treat',
          'description': 'This questionnaire triages and treats patients',
          'status': 'PUBLISHED'
        },
        {
          '_id': '5d1c95a173589a0030d798af',
          'title': 'Test',
          'description': 'Test questionnaire',
          'status': 'DRAFT'
        }
      ]
    })


  it('should render correctly', async () => {
    const component = mount(<ClinicianDashboard state={{ value: 0, totalQuestionnaire: 0, totalPending: 0, totalClose: 0 }} />)
    component.update()
    expect(component).toMatchSnapshot()
  })
})