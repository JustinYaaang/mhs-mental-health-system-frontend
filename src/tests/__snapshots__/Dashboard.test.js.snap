// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyCreator should render correctly 1`] = `
<Dashboard
  classes={
    Object {
      "cardCategory": "Dashboard-cardCategory-4",
      "cardCategoryWhite": "Dashboard-cardCategoryWhite-5",
      "cardTitle": "Dashboard-cardTitle-6",
      "cardTitleWhite": "Dashboard-cardTitleWhite-7",
      "stats": "Dashboard-stats-3",
      "successText": "Dashboard-successText-1",
      "upArrowCardCategory": "Dashboard-upArrowCardCategory-2",
    }
  }
/>
`;
